h2o-core/src/main/java/hex/ModelMetrics.java

```diff
@@ -205,8 +205,8 @@ private final ModelMetrics findMetricsForModel(Key<Model> modelKey) {
       Model model = modelKey.get();
       mm = ModelMetrics.getFromDKV(model, this.frame);
       if (null == mm) {
-        // call score()
-        Frame preds = model.score(this.frame);
```

Comment:
It would be better to use `model.scoreMetrics` which doesn't create the prediction Frame. It would be a little bit more complicated because scoreMetrics assumes adapted Frame.

@seb-h2o I do think this whole class MetricsComparatorForFrame deserves some refactoring. It could look much simpler if we first calculated the missing metrics and then ran the comparator on the precalculated metrics. The fact that the metrics are calculated on demand when they are needed by the compare method is dangerous. It is really easy to introduce a bug by not putting the metrics in cache and re-score again. This approach is really not predictable.

Reply:
agree, was surprised to this being called only by compare method as well.
I can create a Jira ticket for this

Reply:
created PUBDEV-5963